addons/account/views/account_journal_dashboard_view.xml

```diff
@@ -330,14 +330,13 @@
                             </div>
                             <div class="row" t-if="dashboard.number_waiting">
                                 <div class="col overflow-hidden text-start">
-                                    <a type="object" t-if="journal_type == 'sale'" name="open_action"
+                                    <a type="object" name="open_action"
                                     context="{'search_default_open':1, 'search_default_posted':1, 'search_default_partial': 1}" id="account_dashboard_sale_pay_link">
-                                        <t t-out="dashboard.number_waiting"/> Unpaid Invoices
+                                        <span t-if="journal_type == 'sale'" title="Unpaid Invoices"><t t-out="dashboard.number_waiting"/> Unpaid Invoices</span>
                                     </a>
-
-                                    <a type="object" t-if="journal_type == 'purchase'" name="open_action"
-                                    context="{'search_default_open':1, 'search_default_posted':1, 'search_default_partial': 1}" id="account_dashboard_purchase_pay_link">
-                                        <t t-out="dashboard.number_waiting"/> Bills to Pay
+                                    <a type="object" name="action_open_payment_items"
+                                    context="{'search_default_purchases': '1'}">
```

Comment:
The "Invoices to Validate", "Bills to Validate", "Unpaid Invoice" and "Late Invoice" texts are clickable and trigger the open_action method. That method assigns the right filters and domain according to the text that was clicked. However, all those redirect to an account.move view while "Bills to Pay" and "Late Bills" redirect to an account.move.line view. Using the same method would probably involve a lot of changes while using a different method is way clearer. This new method is action_open_payment_items and return an easy action which needs a simple domain change depending on whether you click on one or the other.

In short : Sales is used when clicking on an invoice related text but it is done inside the action instead of passed via the context.